src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/Backup.java

```diff
@@ -0,0 +1,481 @@
+
+    /* Job Parameters Keys **/
+    public static final String PARAM_TIME = "time";
+
+    public static final String PARAM_OUTPUT_FILE_PATH = "output.file.path";
+
+    public static final String PARAM_INPUT_FILE_PATH = "input.file.path";
+
+    public static final String PARAM_DRY_RUN_MODE = "BK_DRY_RUN";
+
+    public static final String PARAM_BEST_EFFORT_MODE = "BK_BEST_EFFORT";
+    
+    /* Jobs Context Keys **/
+    public static final String BACKUP_JOB_NAME = "backupJob";
+
+    public static final String RESTORE_JOB_NAME = "restoreJob";
+
+    public static final String RESTORE_CATALOG_KEY = "restore.catalog";
+
+    /** The semaphore */
+    private static final long SIGNAL_TIMEOUT = 300; // TODO: This should be configurable from the GUI
+    CountDownLatch doneSignal = new CountDownLatch(1);
+
+    /** catalog */
+    Catalog catalog;
+
+    GeoServer geoServer;
+
+    GeoServerResourceLoader resourceLoader;
+
+    GeoServerDataDirectory geoServerDataDirectory;
+
+    XStreamPersisterFactory xpf;
+
+    JobOperator jobOperator;
+
+    JobLauncher jobLauncher;
+
+    Job backupJob;
+
+    Job restoreJob;
+
+    ConcurrentHashMap<Long, BackupExecutionAdapter> backupExecutions = new ConcurrentHashMap<Long, BackupExecutionAdapter>();
+
+    ConcurrentHashMap<Long, RestoreExecutionAdapter> restoreExecutions = new ConcurrentHashMap<Long, RestoreExecutionAdapter>();
+
+    private Integer totalNumberOfBackupSteps;
+
+    private Integer totalNumberOfRestoreSteps;
+
+    /**
+     * A static application context
+     */
+    static ApplicationContext context;
+
+    public Backup(Catalog catalog, GeoServerResourceLoader rl) {
+        this.catalog = catalog;
+        this.geoServer = GeoServerExtensions.bean(GeoServer.class);
+
+        this.resourceLoader = rl;
+        this.geoServerDataDirectory = new GeoServerDataDirectory(rl);
+        
+        this.xpf = GeoServerExtensions.bean(XStreamPersisterFactory.class);
+    }
+
+    /**
+     * @return the jobOperator
+     */
+    public JobOperator getJobOperator() {
+        return jobOperator;
+    }
+
+    /**
+     * @return the jobLauncher
+     */
+    public JobLauncher getJobLauncher() {
+        return jobLauncher;
+    }
+
+    /**
+     * @return the Backup job
+     */
+    public Job getBackupJob() {
+        return backupJob;
+    }
+
+    /**
+     * @return the Restore job
+     */
+    public Job getRestoreJob() {
+        return restoreJob;
+    }
+
+    /**
+     * @return the backupExecutions
+     */
+    public ConcurrentHashMap<Long, BackupExecutionAdapter> getBackupExecutions() {
+        return backupExecutions;
+    }
+
+    /**
+     * @return the restoreExecutions
+     */
+    public ConcurrentHashMap<Long, RestoreExecutionAdapter> getRestoreExecutions() {
+        return restoreExecutions;
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationEvent event) {
+        // load the context store here to avoid circular dependency on creation
```

Comment:
Isn't it possible to avoid the circularity to start with? Like, for example, I see CatalogReader uses the facade to get the current execution by id, with some programmatic logic... wouldn't it be better to put the execution in the ExecutionContext, which is then directly accessible? The same goes for accessing the catalog and so on.